src/datepicker/ngb-date.ts

```diff
@@ -1,6 +1,15 @@
 export class NgbDate {
-  static from(date: {year: number, month: number, day?: number}) {
-    return date ? new NgbDate(date.year, date.month ? date.month : 1, date.day ? date.day : 1) : null;
+  static from(date: {year: number, month?: number, day?: number}) {
+    let nDate = date ? new NgbDate(date.year, date.month ? date.month : 1, date.day ? date.day : 1) : null;
+    if (nDate) {
+      // add 1 to year to prevent any date in last valid javascript Date being picked and causing issues around
+      // September
+      let jDate = new Date(nDate.year + 1, nDate.month - 1, nDate.day);
```

Comment:
Due to how NgbCalendar works for the Gregorian Calendar implementation, yeah we have to. [It converts to a JS Date](https://github.com/ng-bootstrap/ng-bootstrap/blob/3af832995eaf2c85b516b465b8e9c03092b787fe/src/datepicker/ngb-calendar.ts#L37) in `getNext` , `getWeekday` and `getWeekNumber`. The invalid JS Date object is what was causing the infinite loop in the DatepickerService